ui/lib/core/addon/helpers/options-for-backend.js

```diff
@@ -166,10 +178,18 @@ const SECRET_BACKENDS = {
   },
 };
 
-export function optionsForBackend([backend, tab]) {
+export function optionsForBackend([backend, tab, isEngine]) {
   const selected = SECRET_BACKENDS[backend];
+  const selectedEngine = ENGINE_SECRET_BACKENDS[backend];
   let backendOptions;
-
+  if (isEngine) {
+    let tabData =
+      selectedEngine.tabs.findBy('name', tab) ||
+      selectedEngine.tabs.findBy('modelPrefix', tab) ||
+      selectedEngine.tabs[0];
+    backendOptions = assign({}, selectedEngine, tabData);
+    return backendOptions;
```

Comment:
exits out of the function with this return.

Reply:
I wonder if rather than adding the if statement you could do a ternary for `selected`?
```js
const selected = isEngine ? ENGINE_SECRET_BACKENDS[backend] : SECRET_BACKENDS[backend];
```
It looks like the rest of the logic could stay the same.

Reply:
Yes! Much better, thank you.